opentracing-api/src/main/java/io/opentracing/Reference.java

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright 2016 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.opentracing;
+
+/**
+ * A Reference pairs a reference type with a SpanContext referee.
+ *
+ * References are used by Tracer.buildSpan() to describe the relationships
+ between Spans.
+ */
+public class Reference {
```

Comment:
I don't quite understand why we are pulling out this type. Seems simpler to rename Reference.Type to ReferenceType and squash the rest of this class. We could make one of the referenceTypes default.

If there's some compelling reason for making this object, it is likely a good thing for the doc section.